lib/active_admin/filters/active_filter.rb

```diff
@@ -90,16 +90,27 @@ def find_class
         if condition_attribute.klass != resource_class && condition_attribute.klass.primary_key == name.to_s
           condition_attribute.klass
         else
-          assoc = condition_attribute.klass.reflect_on_all_associations.
-            reject { |r| r.options[:polymorphic] }. #skip polymorphic
-            detect { |r| r.foreign_key.to_s == name.to_s }
-          assoc.class_name.constantize if assoc
+          predicate_association.class_name.constantize if predicate_association
         end
       end
 
       def filter
         resource.filters[name.to_sym]
       end
+
+      def related_primary_key
+        if predicate_association
+          predicate_association.active_record_primary_key
+        else
+          related_class.primary_key if related_class
```

Comment:
Should this become an `elsif` for easier reading?

Reply:
Like line 93, since a there were other places where it seemed that dangling if - elsif weren't preferred, I stuck with the if - else. But IMHO I agree that it should just be an if - elsif block